src/xenia/app/emulator_window.cc

```diff
@@ -864,6 +872,26 @@ void EmulatorWindow::OnKeyDown(ui::KeyEvent& e) {
   e.set_handled(true);
 }
 
+void EmulatorWindow::OnMouseDown(ui::MouseEvent& e) {
+  auto now = std::chrono::steady_clock::now();
+  if (std::chrono::duration_cast<std::chrono::milliseconds>(now - last_mouse_click_down).count() < 250) {
+    // avoid multiple mouse down events firing consecutively
+    if (std::chrono::duration_cast<std::chrono::milliseconds>(now - last_mouse_click_up).count() < 250) {
+      // last time mouse was released was no longer than 500ms ago
+      if (std::chrono::duration_cast<std::chrono::milliseconds>(last_mouse_click_up - last_mouse_click_down).count() < 100) {
+        // avoid 'chaining' double clicks, in sequence with 'last achieved' double click
+        ToggleFullscreen();
+      }
+    }
+  } else {
+    last_mouse_click_down = now;
+  }
+}
```

Comment:
1. Please extract it to another function
2. Such if-nesting can be avoided
3. I would create a const for these magic numbers
4. Use linter

Reply:
i've made some changes to cover the above mentioned action items.  based on what I could find in Contributing.md i take it that the linter you are using is `cpplint`, right? used that one to fix all linting issues introduced by these code changes.

Reply:
Nice, looks good to me. One more thing. Please squash commits into one and change commit message to something like this: "[UI] Set fullscreen on double left mouse click"

Reply:
alright, just squashed these two commits into 1; the commit message was changed to "[UI] Set fullscreen on double left mouse click".